pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,399 @@
+
+	flags.BoolVar(&cfg.Schedulable, "schedulable", false, "Control pod schedulability on the node.")
+	flags.BoolVar(&cfg.Evacuate, "evacuate", false, "Migrate all/selected pods on the node.")
+	flags.BoolVar(&cfg.ListPods, "list-pods", false, "List all/selected pods on the node.")
+
+	flags.BoolVar(&cfg.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
+	flags.Var(&cfg.PodSelector, "pod-selector", "Label selector to filter pods on the node. Optional parameters for --evacuate or --list-pods")
+	flags.Var(&cfg.Selector, "selector", "Label selector to filter nodes. Either pass one/more nodes as arguments or use this node selector")
+
+	return cmd
+}
+
+func (cfg *NodeConfig) ValidateParams() []error {
+	errList := []error{}
+	if cfg.Selector.Provided() {
+		if _, err := labels.Parse(cfg.Selector.Value()); err != nil {
+			errList = append(errList, errors.New("--selector=<node_selector> must be a valid label selector"))
+		}
+		if len(cfg.NodeNames) != 0 {
+			errList = append(errList, errors.New("either specify --selector=<node_selector> or nodes but not both"))
+		}
+	} else if len(cfg.NodeNames) == 0 {
+		errList = append(errList, errors.New("either specify --selector=<node_selector> or nodes"))
+	}
+
+	numOps := len(cfg.Operations)
+	if numOps == 0 {
+		errList = append(errList, errors.New("must provide a node operation. Supported operations: --schedulable, --evacuate and --list-pods"))
+	} else if numOps != 1 {
+		errList = append(errList, errors.New("must provide only one node operation at a time"))
+	}
+
+	if cfg.DryRun && !cfg.Evacuate {
+		errList = append(errList, errors.New("--dry-run is only applicable for --evacuate"))
+	}
+
+	if cfg.PodSelector.Provided() {
+		if _, err := labels.Parse(cfg.PodSelector.Value()); err != nil {
+			errList = append(errList, errors.New("--pod-selector=<pod_selector> must be a valid label selector"))
+		}
+	}
+	return errList
+}
+
+// RunManageNode contains all the necessary functionality for the openshift cli manage-node command
+func (cfg *NodeConfig) RunManageNode() error {
+	nameArgs := []string{"nodes"}
+	if len(cfg.NodeNames) != 0 {
+		nameArgs = append(nameArgs, cfg.NodeNames...)
+	}
+
+	r := resource.NewBuilder(cfg.Mapper, cfg.Typer, resource.ClientMapperFunc(cfg.RESTClientFactory)).
+		ContinueOnError().
+		NamespaceParam(cfg.DefaultNamespace).
+		SelectorParam(cfg.Selector.Value()).
+		ResourceTypeOrNameArgs(true, nameArgs...).
+		Flatten().
+		Do()
+	if r.Err() != nil {
+		return r.Err()
+	}
+
+	var err error
+	errList := []error{}
+	nodeCount := 0
+	ignoreHeaders := false
+	err = r.Visit(func(info *resource.Info) error {
+		nodeCount++
+		node, ok := info.Object.(*kapi.Node)
+		if !ok {
+			err = fmt.Errorf("cannot convert input to Node: ", reflect.TypeOf(info.Object))
+			errList = append(errList, err)
+			return nil
+		}
+
+		switch cfg.Operations[0] {
+		case "schedulable":
+			err = cfg.RunSchedulable(node, ignoreHeaders)
+			ignoreHeaders = true
```

Comment:
If you're going to use it, use it consistently.  It seems just as valid for the other operations, though I don't know that I really care whether headers come out multiple times.